test/uk/gov/hmrc/helptosavestub/controllers/NSIControllerSpec.scala

```diff
@@ -89,4 +90,39 @@ class NSIControllerSpec extends UnitSpec with WithFakeApplication {
       status(result) shouldBe BAD_REQUEST
     }
   }
+
+  "Get /nsi-services/account" should {
+    "return a successful status when given an existing nino" in {
+      val request = FakeRequest()
+        .withHeaders(authHeader)
+      val result = NSIController.getAccount(Some("correlationId"), Some("EM000001A"), Some("V1.0"), Some("systemId"))(request)
+      status(result) shouldBe OK
+      val json = contentAsString(result)
+      Json.fromJson[NSIGetAccountByNinoResponse](Json.parse(json)).get shouldBe NSIGetAccountByNinoResponse.bethNSIResponse(Some("correlationId"))
+    }
+
+    "return a 400 when given a nino not found" in {
```

Comment:
it would be nice to have coverage for the different errorMessageIds that can be returned

Reply:
Added more tests to cover all errorMessageIds apart from the two systemId ones, as we currently do not have any validation on that field so these errors are never being triggered.

Reply:
Thanks, if that means there is unused code (e.g. systemId-related errors in NSIErrorResponse) then I reckon it's best to remove it. 2 reasons: it's best to avoid unused code to make the code easier to understand & maintain, and the systemId errors aren't based on the ICD (the errorMessageIds they use aren't sourced from the ICD, unlike the errorMessageIds for other errors which do originate from the ICD).

Reply:
True, I added those systemId errors myself, so I've now removed them, thanks for spotting.